packages/dds/tree/src/simple-tree/node-kinds/object/objectNode.ts

```diff
@@ -639,3 +667,120 @@ export function createUnknownOptionalFieldPolicy(
 		);
 	};
 }
+
+/**
+ * Transforms data under an Object schema.
+ * @param data - The tree data to be transformed. Must be a Record-like object.
+ * @param schema - The schema associated with the value.
+ */
+function objectToFlexContent(
```

Comment:
this logic and below was just moved out of UnhydratedFlexTreeFromInsertable. Same for the other node kinds.